c10s: create new podman files for c10s

vbenes opened the pull request:
* We don't have epel repo so using files from Fedora40 where needed.
* For tox and virtualenv we need to use pip to be compatible for the same lack of epel reasons.
* It seems /etc/systemd/journald.conf is not present either so RateLimits added directly (not sure here if not too invasive)
* We don't need nispor as it should be part of nmstate now.

vbenes committed:
c10s: create new podman files for c10s

* We don't have epel repo so using copr from openvswitch rpms.
* /etc/systemd/journald.conf is not present either so RateLimits
  added directly
* Some packages still missing so we need to skip tests requiring
  them in c10s for now.

Signed-off-by: Vladimír Beneš <vbenes@redhat.com>

cathay4t commented:
1. We still need nispor CLI `npc` in test.
2. Please expand `run-tests.sh` with `--el10` argument. And also `run-tests-in-nmci.sh`. This would show whether the container works or not.
3. You may tag your local build as `quay.io/nmstate/c10s-nmstate-build:latest`, so runtest.sh will use local built result. `podman build -f Dockerfile.c10s-nmstate-build -t quay.io/nmstate/c10s-nmstate-build:latest`

vbenes committed:
automation: add c10s into run-tests scripts

We have two scripts, run-tests.sh which is executing automation and
NMCI's specific run-tests-in-nmci.sh which calls the previous one.
Both scripts have now el10 option.

Signed-off-by: Vladimír Beneš <vbenes@redhat.com>

cathay4t commented on automation/run-tests.sh:
```diff
@@ -20,6 +20,7 @@ FEDORA_IMAGE_DEV="quay.io/nmstate/fed-nmstate-dev:latest"
 RAWHIDE_IMAGE_DEV="quay.io/nmstate/fed-nmstate-dev:rawhide"
 CENTOS_8_STREAM_IMAGE_DEV="quay.io/nmstate/c8s-nmstate-dev"
 CENTOS_9_STREAM_IMAGE_DEV="quay.io/nmstate/c9s-nmstate-dev"
+CENTOS_10_STREAM_IMAGE_DEV="quay.io/nmstate/c10s-nmstate-build"
```
Should be `quay.io/nmstate/c10s-nmstate-dev`

vbenes replied:
done

cathay4t commented on packaging/Dockerfile.c10s-nmstate-dev:
```diff
@@ -0,0 +1,35 @@
+COPY network_manager_keyfile.conf \
+     /etc/NetworkManager/conf.d/96-keyfile.conf
+
+# Fedora container has no /etc/systemd/journald.conf file in systemd rpm
```
RHEL/CentOS stream 10

vbenes replied:
done as well

vbenes committed:
CI: add basic nm_stable bits for c10s CI

Signed-off-by: Vladimír Beneš <vbenes@redhat.com>

cathay4t committed:
rpm: Fix %cargo_build error on c10s

The %cargo_build in CentOS stream 10 does not allows extra argument,
hence we skip the `--ignore-rust-version` for now.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t reviewed (approved):
The OVS tests are failing, but good enough for first approach of c10s tests.
